components/SearchContainer.js

```diff
@@ -0,0 +1,88 @@
+import styled from "styled-components";
+import TextEntry from "./TextEntry";
+import Button from "./Button";
+import { useRouter } from "next/router";
+
+export default function SearchContainer() {
+  const router = useRouter();
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+    const formdata = new FormData(event.target);
+    const data = Object.fromEntries(formdata);
+    const { thema, source, publishedAt, country } = data;
+    const response = await fetch(
+      `/api/search?thema=${thema}&source=${source}&when=${publishedAt}&country=${country}`
+    );
+    const results = await response.json();
+    localStorage.setItem("articles", JSON.stringify(results));
+    router.push("/");
+  }
+  return (
+    <StyledSearchContainer onSubmit={handleSubmit}>
+      <TextEntry
+        label="thema"
+        id="thema"
+        name="thema"
+        placeholder="eg. politics"
+      />
+      <TextEntry
+        label="source"
+        id="source"
+        name="source"
+        placeholder="eg. nytimes"
+      />
+      <TextEntry
+        label="published at"
+        id="publishedAt"
+        name="publishedAt"
+        placeholder="eg. 2022-10-01"
+      />
+      <TextEntry
+        label="country"
+        id="country"
+        name="country"
+        placeholder="eg. england"
+      />
+      <StyledArticle>
```

Comment:
Article tags are not used for formulars / inputs. You could use sections or divs.

Reply:
Thank you!